cmd/podman/pods/create.go

```diff
@@ -240,25 +228,10 @@ func create(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	podSpec.InfraImage = customInfraImage
+
 	if createOptions.Infra {
-		rawImageName = img
-		if !infraOptions.RootFS {
-			curr := infraOptions.Quiet
-			infraOptions.Quiet = true
-			name, err := containers.PullImage(imageName, infraOptions)
-			if err != nil {
-				fmt.Println(err)
-			}
-			imageName = name
-			infraOptions.Quiet = curr
-		}
-		podSpec.InfraImage = imageName
-		if infraOptions.Entrypoint != nil {
-			createOptions.InfraCommand = infraOptions.Entrypoint
-		}
-		podSpec.InfraContainerSpec = specgen.NewSpecGenerator(imageName, false)
-		podSpec.InfraContainerSpec.RawImageName = rawImageName
-		podSpec.InfraContainerSpec.NetworkOptions = podSpec.NetworkOptions
+		podSpec.InfraContainerSpec = specgen.NewSpecGenerator(customInfraImage, false)
 		err = specgenutil.FillOutSpecGen(podSpec.InfraContainerSpec, &infraOptions, []string{})
```

Comment:
This is what you would remove if you wanted to move filling out spec to further down the stack.